stl/inc/xutility

```diff
@@ -208,6 +208,15 @@ struct _Get_rebind_alias<_Ty, _Other, void_t<typename _Ty::template rebind<_Othe
     using type = typename _Ty::template rebind<_Other>;
 };
 
+#if _HAS_CXX23
+template <class _Raw, class _Source>
+_NODISCARD constexpr auto&& _Forward_as(auto&& _Val) noexcept {
+    _STL_INTERNAL_STATIC_ASSERT(is_same_v<_Source&, decltype(_Val)&>);
+    using _Cooked = typename _Add_qualifiers<_Source&&>::template _Apply<_Raw>;
+    return (_Cooked) _Val; // C-style cast used to bypass access control

```

Comment:
IMO this is just the standard being too restrictive. `static_cast` from private subclass to your own class should be valid, since the rule forbidding it is basically [[conv.ptr]/3](https://eel.is/c++draft/conv.ptr#3) forbidding "converting from pointer/reference to derived class object to inaccessible base class", and [[expr.cast]/4.6](https://eel.is/c++draft/expr.cast#4.6) confirms that C-style casts can do this.

However, inside the base class scope, probably it should be okay for a base class member function to get the base class subobject of the derived class, no matter what kind of inheritance it is...? I'm not sure about the consequence about this. In other word, given `struct D : private B`, `B` is an inaccessible base of `D`, that's not going to change; however, inside `B` itself, `B` should not be inaccessible anymore.

Reply:
@Mick235711 I think we can file a CWG issue at https://github.com/cplusplus/CWG/issues.
How about such modification (adding to [conv.ptr]/3):
> ... If `B` is an inaccessible ([class.access]) or ambiguous ([class.member.lookup]) base class of `D`**, where the accessibility is checked in the context the conversion occurs in**, a program that necessitates this conversion is ill-formed. ...